subsys/net/lib/rest_client/src/rest_client.c

```diff
@@ -197,7 +197,8 @@ static int rest_client_sckt_connect(int *const fd,
 		/* Check if timeout has already elapsed */
 		if (time_used >= *timeout_ms) {
 			LOG_WRN("Timeout occurred during DNS query");
-			return -ETIMEDOUT;
+			ret = -ETIMEDOUT;
+			goto clean_up;
```

Comment:
There's another timeout check around line 245, that should probably be handled in the same way.

Reply:
Thanks, added.